debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java

```diff
@@ -291,6 +244,16 @@ else if (oplogStart.getTime() > 0) {
                         break;
                     }
                 }
+
+                try {
+                    oplogContext.getOffset().cursorEvent(cursor);
```

Comment:
My naive attempt at getting heartbeats to work from the cursor `resumeToken`. Also note that I moved this to happen even when `event` is undefined. It probably should be guarded to only first the heartbeat event wen the cursor has a non-null `resumeToken`.

Reply:
I'm not sure I understand this comment... Shouldn't the cursor always have a non-null token (unless it was closed)?  Also what do you mean by "undefined" event?

Reply:
[This part of the spec](https://github.com/mongodb/specifications/blob/master/source/change-streams/change-streams.rst#updating-the-cached-resume-token) makes me believe it can still be `null`:
> If an empty batch was returned and a `postBatchResumeToken` was included, cache it.
> [...]
> If it's the last document in the batch and a `postBatchResumeToken` is included, cache it.

Additional context [here](https://www.mongodb.com/docs/manual/changeStreams/#resume-tokens) and [here](https://www.mongodb.com/docs/manual/reference/command/getMore/#mongodb-dbcommand-dbcmd.getMore).

Furthermore, since we support so many versions of Mongo, we cannot rule out things like [SERVER-63772: Post Batch Resume token not sent on initial batch from per shard cursor change stream](https://jira.mongodb.org/browse/SERVER-63772)